flight1handler.go

```diff
@@ -90,6 +90,16 @@ func flight1Generate(c flightConn, state *State, cache *handshakeCache, cfg *han
 		extensions = append(extensions, &extension.ServerName{ServerName: cfg.serverName})
 	}
 
+	if cfg.sessionStore != nil {
+		addr := c.RemoteAddr().String()
+		if s := cfg.sessionStore.GetByAddr(addr); s != nil {
```

Comment:
Selecting session by remote address sounds improper.
Server can be multiplexed and nameserver may round-robin them.

Reply:
What about use the `cfg.serverName` to fetch stored session?

Reply:
Maybe better to check the implementation of the other libraries like openssl.

Reply:
@at-wat What about use both `cfg.serverName` and `c.RemoteAddr`?

If the client have set the `cfg.serverName`, use it as key. Otherwise, use the remote address.

Reply:
I'm still not sure what is the correct and secure key of the session.
Finding implementations in other tls/dtls libs or any other documentation would be better.

Reply:
https://github.com/nss-dev/nss/blob/9dab43371d4d924419523e18ba84f02804880533/lib/ssl/sslnonce.c#L253-L308
In NSS (used by Firefox/Chrome, etc), TLS session is reused only if all of IP address, port, server name (mandatory) and proxy peer ID (if used) are matched.